src/EventGridEmulator/Network/TrustingDevelopmentCertificateHttpClientHandler.cs

```diff
@@ -0,0 +1,11 @@
+namespace EventGridEmulator.Network;
+
+public class TrustingDevelopmentCertificateHttpClientHandler : HttpClientHandler
+{
+    private TrustingDevelopmentCertificateHttpClientHandler()
+    {
+        this.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+    }
+
+    public static TrustingDevelopmentCertificateHttpClientHandler Instance { get; } = new TrustingDevelopmentCertificateHttpClientHandler();
```

Comment:
We can leverage dependency injection as we did for all the other classes and not having to manually create a singleton. The `Program.cs` would look like this (specifically the first and last lines have changed):

```csharp
builder.Services.AddSingleton<TrustingDevelopmentCertificateHttpClientHandler>();
builder.Services.AddHttpClient(SubscriberConstants.HttpClientName, SubscriberConstants.ConfigureHttpClient)
    .AddHttpMessageHandler<SubscriberPolicyHttpMessageHandler>()
    .ConfigurePrimaryHttpMessageHandler<TrustingDevelopmentCertificateHttpClientHandler>();
```